src/mpid/ch4/src/ch4_impl.h

```diff
@@ -1146,4 +1146,38 @@ MPL_STATIC_INLINE_PREFIX MPI_Op MPIDIU_win_acc_get_op(int index)
         return MPIR_Op_builtin_get_op(index);
     }
 }
+
+MPL_STATIC_INLINE_PREFIX bool MPIDIG_rma_recv_am_always_poll(void)
+{
+    bool always_poll = false;
+    int interval = MPL_atomic_fetch_add_int(&MPIDIG_global.rma_recv_am_poll_interval, 1);
+
+    if (MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL > 0) {
+        /* User explicitly controls the polling frequency */
+        always_poll =
+            ((interval + 1) % MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL == 0) ? true : false;
+    } else if (MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL == -1) {
+        /* Runtime control is on.
+         * - Always poll if RMA receive AM is active
+         * - Otherwise poll with longer interval */
+        if (MPL_atomic_load_int(&MPIDIG_global.rma_recv_am_active)) {
+            always_poll = true;
+        } else {
+            always_poll =
+                ((interval + 1) % MPIR_CVAR_CH4_RMA_RECV_AM_PROGRESS_INTERVAL_LONG ==
+                 0) ? true : false;
+        }
+    } else {
+        /* User explicitly disables polling */
+        always_poll = false;
+    }
+
```

Comment:
It is not completely disable polling in flushes. We still poll with certain interval

Reply:
I assume if this returns true and the INTERVAL CVAR is set to `0`, then the polling is essentially disabled, right?

Reply:
Returning false will only disable poll for the calling flush. It needs to internally check whether the interval is met. If so, it needs to return true -- meaning the calling flush will poll progress.

Reply:
Let's back up a second. I thought that the goal was to completely disable AM progress when the application (OSHMPI) doesn't need it, right? And right now, AM progress is called during every progress loop iteration? What don't we just add a CVAR to disable it and avoid this interval discussion for right now? Or is the interval-based polling also a critical piece for this release?

I remember a discussion about calling internal abort if an active message is issued on a window that expects only native operations. That sounds OK to me, too.

Reply:
Oh, I guess the problem is the CVAR is global and not per-window...

Reply:
The interval is needed for correctness. If we never poll AM, even the given process is entirely doing native RMA, the other process may still issue am rma targeting the given process, and it will stall due to lack of progress.

Reply:
Right. Without extra hints, we cannot know whether any remote process may issue AM-based RMA to the given process. With interval, it allows the given process to poll progress with a long interval (or low frequency) at beginning, and change to a shorter interval (more frequent, essentially interval=1) polling once it sees any incoming AM-based RMA. 

This way will allow OSHMPI to skip the unnecessary polling for incoming RMA AM in most fast cases (i.e., contig put/get, atomics), while still making progress when an unusual strided put/get (AM-based PUT/GET in MPI) is issued.